src/main/java/duke/Commands.java

```diff
@@ -0,0 +1,86 @@
+package duke;
+
+
+import java.util.Scanner;
+import duke.parser.Parser;
+import duke.storage.Storage;
+import duke.task.ItemList;
+import duke.ui.UI;
+
+
+/**
+ * The Command Class encapsulates the functions to check the input string with the available commands.
+ */
+public class Commands {
+    /**
+     * The Enum Class encapsulates all the available commands.
+     */
+    public enum CommandType {
+        MARK, UNMARK, LIST, EVENT, DEADLINE, TODO, BYE, DELETE, FIND
+
+    }
+
+
+    /**
+     * This method Run the Scanner to begin taking inputs from user, and check to see which commands to run.
+     */
+    public static void run(Storage storage) {
+        Scanner sc = new Scanner(System.in);
+        ItemList items = storage.getItems();
+
+        boolean isRunning = true;
+        do {
+            if (!sc.hasNextLine()) {
+                break;
+            }
+            try {
+                String line = sc.nextLine();
+                Parser parser = new Parser(line);
+                String command = parser.getCommand();
+                CommandType given = CommandType.valueOf(command);
```

Comment:
Maybe can be more specific here in terms of variable naming, instead of just `given`, can put `commandGiven`?
```suggestion
                CommandType commandGiven = CommandType.valueOf(command);
```